### ISSUE TITLE
Network/Error handling delegate

### ISSUE THREAD

**Issue by gouen95** [Giphy/giphy-ios-sdk-ui-example#107]
Would like to see an error handling delegate so that I know that content is somehow failed to retrieve and provided retry capabilities to user.

**Comment by cgmaier** [comment]
hey @gouen95 thanks for the suggestion. are you using a `GiphyGridController`-based solution, or the `GiphyViewController`?

**Comment by gouen95** [comment]
I'm using the GipgyGridController solely just to display GIFs, but the problem I faced is if there's no internet or somehow Giphy failed to retrieve data, I wouldn't have a proper way to tell user what's wrong. What I've done is a custom implementation involving timer and timeouts. 

One good thing is where I searched for gibberish such as 'gameigoaejio34aweda', GiphyGridController would return me 0 item count which I think is perfectly OK.
